doc/rst/developer/bestPractices/DCO.rst

```diff
@@ -0,0 +1,131 @@
+add your 'Signed-off-by' line to your commit message.  For an
+interactive commit like the following:
+
+    git commit -s
+
+you should see the 'Signed-off-by' line in your editor when it brings
+up the buffer representing the commit message.  If you supply the
+commit message on the command-line, the 'Signed-off-by' line will be
+automatically added for you.
+
+    git commit -s -m "This is my commit message"
+
+This approach has the advantage of being fairly straightforward and
+requiring no configuration, but the downside that you need to remember
+to use the flag with each commit.
+
+
+#. Using a git commit template
+
+This approach uses a template file that contains your signature for
+interactive commit messages.  It has the advantage of being fairly
+straightforward and making your signature visible to you when you
+commit.  However, a downside is that it doesn't apply your signature
+to git commits that skip the interactive editing of commit messages,
+like `git revert` or `git commit -m ...`.
+
+To take this approach, create a file (say, `~/.gitmessage`) that will
+form the basis for any interactive commit messages, containing your
+signature line:
+
+.. code-block:: bash
+
+     ---
+     Signed-off-by: Random J Developer <random@developer.example.org>
+
+Then, tell your git configuration to open this file for any new
+interactive commit messages using one of the following forms.  To set
+this up for the current repository only:
+
+.. code-block:: bash
+
+     git config commit.template ~/.gitmessage
+
+To set it up across repositories via your `~/.gitconfig` file:
+     
+.. code-block:: bash
+
+     git config --global commit.template ~/.gitmessage
+     
+
+#. Using a git commit hook
+
+This approach uses a script to automatically add your signature line
+to new commits within a given repository if it isn't found in the
+commit message.  Relative to the previous approach, it has the
+advantage of being used more consistently for different styles of
+committing, but the slight downsides of needing to be set up for each
+repository and making the fact that you're signing your commits less
+visible to you / easier to forget about.
+
+To take this approach, copy the script from
+`$CHPL_HOME/util/misc/pre-commit-msg-hook` to
+`$CHPL_HOME/.git/hooks/commit-hook`.
+
+You can also use variations on this script to exit with an error if
+the signature line is missing rather than adding it automatically.
+   
+
+Troubleshooting DCOs
+--------------------
+    
+If you have authored a commit that is missing its 'Signed-off-by'
+line, you can amend your commits and push them to GitHub.
+
+    git commit --amend --signoff
+
+If you've pushed your changes to GitHub already you'll need to force
+push your branch after this with `git push -f`.
+
+If your Pull Request fails the DCO check, it will be necessary to fix
+the entire commit history for the PR. Best practice is to squash the
```

Comment:
Adding onto this, GitHub's "squash and merge" removes the DCO text from the only commit (the merge commit). 

External users shouldn't have to worry about this, but I think it's worth mentioning somewhere that "squash and merge" button should not be used (unless we don't care about DCO check failing _after_ the PR is merged).

Reply:
I'm adding a paragraph to the section "Reviewer responsibilities":

```
  Note that using GitHub's "squash and merge" feature will effectively
  drop all DCO signature lines from the pull request, and so should
  not be used on external commits (or potentially even for commits
  from the core team?) in order to preserve the signed nature of the
  commits.
```

Reply:
(of course, whether or not anyone ever reads and remembers it is another question... :) ).

Reply:
True. Either way, it's better written down than implied.